Allow remark plugins to affect getImage call for .md files (#9566)
withastro/astro: .changeset/calm-socks-shake.md

---
"@astrojs/markdown-remark": minor
"astro": minor
---

> Allow remark plugins to add hProperties to imgs that will be included in the getImage call if they are optimized.

Reviewer: @OliverSpeir , could you instead describe what the user-facing benefit/feature is here? What does this actually do or unlock? Sounds like it improves using remark plugins? How so? to do what? That's the kind of changeset message that would be helpful to have here.

> Users can now create remark plugins that will be able to pass options for how images in `.md` files will be optimized 

Reviewer: ```suggestion
Allows remark plugins to pass options specifying how images in `.md` files will be optimized
```
Love this change, @OliverSpeir ! Something like this, maybe?
